core-backend/sandbox/src/runtime.rs

```diff
@@ -39,37 +41,41 @@ pub(crate) fn as_i64(v: Value) -> Option<i64> {
     }
 }
 
-pub(crate) fn as_u64(v: Value) -> Option<u64> {
-    as_i64(v).map(|v| v as u64)
+pub(crate) fn caller_host_state_take<Ext>(
+    caller: &mut Caller<'_, HostState<Ext, DefaultExecutorMemory>>,
+) -> State<Ext, DefaultExecutorMemory> {
+    caller
+        .data_mut()
+        .take()
+        .unwrap_or_else(|| unreachable!("host_state must be set before execution"))
```

Comment:
we can just `expect` there. It would also be nice to mark the function with `track_caller`.

Reply:
agree on track caller for similar fns